src/pexprs-madlibs.js

```diff
@@ -0,0 +1,209 @@
+'use strict';
+
+// --------------------------------------------------------------------
+// Imports
+// --------------------------------------------------------------------
+
+var common = require('./common');
+var pexprs = require('./pexprs');
+
+// --------------------------------------------------------------------
+// Operations
+// --------------------------------------------------------------------
+
+pexprs.PExpr.prototype.generateExample = common.abstract;
+
+pexprs.any.generateExample = function(grammar, examples, inSyntacticContext, actuals) {
+  return {example: String.fromCharCode(Math.floor(Math.random() * 255))};
+};
+
+function categorizeExamples(examples) {
+  var examplesNeeded = examples.filter(function(example) {
+                                  return example.hasOwnProperty('examplesNeeded');
+                                })
+                               .map(function(example) { return example.examplesNeeded; });
+  examplesNeeded = [].concat.apply([], examplesNeeded)
+                            .filter(function(value, index, arr) {
+                              return arr.indexOf(value) === index;
+                            });
+
+  var successfulExamples = examples.filter(function(example) {
+                                      return example.hasOwnProperty('example');
+                                    })
+                                   .map(function(item) { return item.example; });
+  // could check examples here, here's an example that's invalid
+
+  var needHelp = examples.some(function(item) { return item.needHelp; });
+
+  return {
```

Comment:
It would be nice to have a comment here explaining what these fields mean. After reading the code, I understand `examplesNeeded`, but don't yet understand `needHelp`.